packages/playground/remote/src/lib/worker-caching.ts

```diff
@@ -0,0 +1,92 @@
+import { isURLScoped } from '@php-wasm/scopes';
+
+export class WorkerCache {
+	readonly cacheNamePrefix = 'playground-cache';
+
+	private cacheName: string;
+
+	constructor(cacheVersion: string) {
+		this.cacheName = `${this.cacheNamePrefix}-${cacheVersion}`;
+	}
+
+	addCache = async (key: Request, response: Response) => {
+		const clonedResponse = response.clone();
+		const cache = await caches.open(this.cacheName);
+		await cache.put(key, clonedResponse);
+	};
+
+	getCache = async (key: Request) => {
+		const cache = caches.open(this.cacheName);
+		return await cache.then((c) => c.match(key, { ignoreSearch: true }));
+	};
+
+	isValidHostname = (url: URL) => {
+		/**
+		 * The development environment uses Vite which doesn't work offline because it dynamically generates assets.
+		 * Check the README for offline development instructions.
+		 */
+		if (
+			url.href.startsWith('http://127.0.0.1:5400/') ||
+			url.pathname.startsWith('/website-server/')
+		) {
+			return false;
+		}
+
+		/**
+		 * Scoped URLs are requests made to the PHP Worker Thread.
+		 * These requests are not cached because they are not static assets.
+		 */
+		if (isURLScoped(url)) {
+			return false;
+		}
+
+		/**
+		 * Don't cache PHP files because they are dynamic.
+		 */
+		if (url.pathname.endsWith('.php')) {
+			return false;
+		}
+		return ['playground.wordpress.net', '127.0.0.1', 'localhost'].includes(
```

Comment:
@adamziel you mentioned that there is a way to not hardcode `'playground.wordpress.net'` but I couldn't find it.

I would like to keep this limited to Playground.WordPress.net for now to be safe, so a hardcoded value isn't a blocker.

Reply:
Perhaps the target hostname should be an argument to the WorkerCache constructor. Workers have access to `self.location.hostname`, and that value could be passed to the constructor.

Reply:
Good point @brandonpayton. I made the change in [b888b97](https://github.com/WordPress/wordpress-playground/pull/1535/commits/b888b97cfbea3e0acf0c69f8192e15c75b22fcb6).